RecoParticleFlow/PFClusterProducer/plugins/PFClusterFromHGCalMultiCluster.cc

```diff
@@ -14,43 +17,69 @@ void PFClusterFromHGCalMultiCluster::buildClusters(const edm::Handle<reco::PFRec
   const auto& hgcalMultiClusters = *clusterH_;
   auto const& hits = *input;
 
+  const auto& tracksters = *trackstersH_;
+
   // for quick indexing back to hit energy
   std::unordered_map<uint32_t, size_t> detIdToIndex(hits.size());
   for (uint32_t i = 0; i < hits.size(); ++i) {
     detIdToIndex[hits[i].detId()] = i;
   }
 
+  int iMultiClus = -1;
+
   for (const auto& mcl : hgcalMultiClusters) {
+    iMultiClus++;
+
+    // Filter using trackster PID
+    if (filterByTracksterPID_) {
+      float probTotal = 0.0f;
+
+      for (int& cat : filter_on_categories_) {
```

Comment:
just `int` (or `auto`) - better to handle primitive types by value rather than by reference